### PR TITLE
Extract EA Home topics bar into it's own component

### PR DESCRIPTION
**Pull request by Stvad**
We're reusing it on WakingUP side

This also adds description on hover for tags (disabled by default)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206284705489328) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/common/HomeTagBar.tsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+export type TopicsBarTab = {
+  _id: string,
+  name: string,
+  shortName?: string | null,
+  slug?: string,
+  description?: TagFragment_description | null,
+}
+
+const HomeTagBar = (
```
Comment (s-cheng) [inline packages/lesswrong/components/common/HomeTagBar.tsx:144]: Can you add a docstring to explain what the purpose of this component is?

File: packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx [hunk 2]
```diff
@@ -162,59 +50,9 @@ const EAHomeMainContent = ({FrontpageNode, classes}:{
...
-  const [leftArrowVisible, setLeftArrowVisible] = useState(false)
-  const [rightArrowVisible, setRightArrowVisible] = useState(true)
-  const navigate = useNavigate();
-  const { location, query } = useLocation()
-  const { captureEvent } = useTracking()
-  const activeCoreTopic = useMemo(
-    () => coreTopics?.find(t => t._id === activeTab._id),
-    [coreTopics, activeTab]
-  );
+  const activeCoreTopic = activeTab._id === frontpageTab._id ? null : activeTab as TagDetailsFragment
```
Comment (s-cheng) [inline packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx:55]: Please add a code comment when using `as` to explain why you need it (or refactor to remove the usage if possible)

**Review by s-cheng — CHANGES_REQUESTED** [review]
Returning to the "Frontpage" tab after viewing another tab causes an infinite loop. Other than that this looks good, thanks!

Let me know when you've addressed the issues and I'll do a quick review 👍

File: packages/lesswrong/components/common/HomeTagBar.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+    backgroundColor: theme.palette.grey[200],
+    color: theme.palette.grey[900],
+    fontFamily: theme.typography.fontFamily,
+    fontSize: 14,
+    lineHeight: '23px',
+    fontWeight: '500',
+    padding: '4px 8px',
+    borderRadius: 6,
+    cursor: 'pointer',
+    marginBottom: 8,
```
[the remark this answers is not in the document]
Comment (Stvad) [reply packages/lesswrong/components/common/HomeTagBar.tsx:103]: Margin is there so the description preview tooltip is not flush with the item when that's enabled. Is there by chance a better way to achieve it?

File: packages/lesswrong/components/common/HomeTagBar.tsx [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+      },
+    },
+  },
+  givingSeasonTab: {
+    backgroundColor: theme.palette.givingPortal[800],
+    color: theme.palette.givingPortal[1000],
+    '&:hover': {
+      backgroundColor: theme.palette.givingPortal[900],
+    },
+  },
```
[the remark this answers is not in the document]
Comment (Stvad) [reply packages/lesswrong/components/common/HomeTagBar.tsx:133]: ah, I cherry picked that commit (removing giving season stuff), but probably copied styles earlier? this is unused  - will clean it up. 

conflict is just coming from components list

File: packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx (hunk 2 above)
Comment (Stvad) [reply packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx:55]: will do, this is an artifact of us not having the tag collection here anymore. plausibly it's cleaner to just fetch this using `useSingle`

File: packages/lesswrong/components/common/HomeTagBar.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/components/common/HomeTagBar.tsx:103]: You can pass `popperClassName` in to `LWTooltip` to style it - setting a `marginTop: 8` there seems to do it

File: packages/lesswrong/components/common/HomeTagBar.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (s-cheng) [reply packages/lesswrong/components/common/HomeTagBar.tsx:133]: Thanks!

**Comment by Stvad** [comment]
> Returning to the "Frontpage" tab after viewing another tab causes an infinite loop

Oh fun, after extraction we've been only testing the Waking UP usage of this and it doesn't happen for us 😅. Investigating why this is happening now!

File: packages/lesswrong/components/common/HomeTagBar.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Stvad) [reply packages/lesswrong/components/common/HomeTagBar.tsx:103]: nice, Thank you!

File: packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx (hunk 2 above)
Comment (Stvad) [reply packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx:55]: Updated to do that, though not quite sure how to test spotlight stuff

**Comment by Stvad** [comment]
oh, requested re-review too early, still have a few things to address 🤔

**Comment by Stvad** [comment]
Ok, it should be good now!

File: packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx [hunk 5]
```diff
@@ -162,59 +51,14 @@ const EAHomeMainContent = ({FrontpageNode, classes}:{
...
-        const topic = coreTopics.find(t => t._id === topicId)
-        if (topic) {
-          tabs.push(topic)
-        }
-      })
-    }
-    return tabs
-  }, [coreTopics])
-  
+  const frontpageTab = useMemo(() => ({_id: '0', name: FRONTPAGE_TAB_NAME}), [])
```
Comment (s-cheng) [inline packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx:54]: This doesn't need to be in the component right? Could we just leave it where it was in the code above?

Comment (Stvad) [reply packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx:54]: I think I needed it in the component part way through the refactoring or smth, but yeah moving it out would be simpler vs using useMemo

File: packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx (hunk 2 above)
Comment (s-cheng) [reply packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx]: Oh sorry! Yeah I didn't quite notice the spotlight stuff before - I need to check with our content team to see if we still need this (we've never used it). I would prefer not having an extra `useSingle` every time you change tabs if we can avoid it, so either we remove that bit of code or we update the topic bar component to return the needed field (`introSequenceId`) and then we can remove the `useSingle`

They should get back to me tomorrow. Also if you prefer, you can just hand this PR off to me and I can wrap it up when they reply. At this point the remaining updates are EAF-specific things so I'm happy to take responsibility for them :)

Comment (Stvad) [reply packages/lesswrong/components/ea-forum/EAHomeMainContent.tsx]: > I would prefer not having an extra useSingle

My hope is that this mostly hits the cache bc we've already loaded the relevant things, but I'm not sure how caching actually works though.

Another option is to make the `onTagSelectionUpdated`/ `activeTab` be `TopicsBarTab | TagFragment` and get back to casting (or even let TS derive the type). Casting is valid here bc all those objects are actually tab objects, but relying on that ties us to the internal implementation of the HomeTagBar.

getting rid of spotlight also works if that's an option ofc 😛.

> They should get back to me tomorrow. Also if you prefer, you can just hand this PR off to me and I can wrap it up when they reply. At this point the remaining updates are EAF-specific things so I'm happy to take responsibility for them :)

yeah, feel free to update this accordingly. Thanks for your help getting this in better shape 🙂

**Comment by Stvad** [comment]
Found a small bug when testing on WakingUP side (this is happening on EA forum rn as-well):
- navigate to 404 page directly (can be other pages e.g. tag page)
- go to / with on-site navigation - offsets won't be recalculated and scroll won't work. 

pushed a fix for it here

**Comment by Stvad** [comment]
Ok, actually linter is telling me my solution is bad 😛

**Comment by Stvad** [comment]
Removed it for now, even though it works (but I guess by accident? thinking about a better way to do that)

**Comment by s-cheng** [comment]
@Stvad Thanks! I just looked at the branch and it looks fixed now, is that right?

I didn't hear back from the content team so I will comment out the spotlight-related code here, so we don't need to deal with it now.

**Review by s-cheng — APPROVED** [review]

**Comment by s-cheng** [comment]
@Stvad Thanks so much for your work on this PR - looks good from my end! I'm not sure if you are still fixing bugs here, so I will let you merge it in when you're ready. Let me know if you need anything else from me.

**Comment by Stvad** [comment]
Thanks! The navigation bug is fixed now, so merging this!

**Pull request merged**